apps/OboeTester/app/src/main/java/com/mobileer/oboetester/RoundTripLatencyActivity.java

```diff
@@ -165,30 +201,287 @@ public void start() {
+                           int actualBufferBursts,
+                           int capacityInBursts,
+                           boolean isMMapExclusive) {
+            int result = RESULT_UNDEFINED;
+            mMessage = "analyzing";
+            int nextBursts = -1;
+            if (!isMMapExclusive) {
+                mMessage = "skipped, not MMAP Exclusive";
+                return RESULT_OK;
+            }
+            switch (mState) {
+                case STATE_MEASURE_LOW:
+                    mLowBufferLatency = latencyFrames;
+                    mLowBufferBursts = actualBufferBursts;
+                    // Now we measure the high side.
+                    mHighBufferBursts = capacityInBursts;
+                    nextBursts = mHighBufferBursts;
+                    mMessage = "checked low bufferSize";
+                    mState = STATE_MEASURE_HIGH;
+                    break;
+                case STATE_MEASURE_HIGH:
+                    mMessage = "checked high bufferSize";
+                    mHighBufferLatency = latencyFrames;
+                    mHighBufferBursts = actualBufferBursts;
+                    if (measureLatencyLinearity(mLowBufferBursts,
+                            mLowBufferLatency,
+                            mHighBufferBursts,
+                            mHighBufferLatency) > MAX_ALLOWED_DEVIATION) {
+                        mMiddleBufferBursts = (mHighBufferBursts + mLowBufferBursts) / 2;
+                        nextBursts = mMiddleBufferBursts;
+                        mState = STATE_MEASURE_MIDDLE;
+                    } else {
+                        result = RESULT_OK;
+                        mMessage = "DSP position looks good";
+                        mState = STATE_DONE;
+                    }
+                    break;
+                case STATE_MEASURE_MIDDLE:
+                    mMiddleBufferLatency = latencyFrames;
+                    mMiddleBufferBursts = actualBufferBursts;
+                    // Check to see which side is bad.
+                    if (confidence < 0.5) {
+                        // We may have landed on the DSP so we got a scrambled result.
+                        result = RESULT_DISCONTINUITY;
+                        mMessage = "on top of DSP!";
+                        mState = STATE_DONE;
+                    } else  {
+                        double deviationLow = measureLatencyLinearity(
+                                mLowBufferBursts,
+                                mLowBufferLatency,
+                                mMiddleBufferBursts,
+                                mMiddleBufferLatency);
+                        double deviationHigh = measureLatencyLinearity(
+                                mMiddleBufferBursts,
+                                mMiddleBufferLatency,
+                                mHighBufferBursts,
+                                mHighBufferLatency);
+                        if (deviationLow >= deviationHigh) {
```

Comment:
What about the case where both sides are bad? Do we have any indication of this?